e3db/config.py

```diff
@@ -35,12 +35,18 @@ def __init__(self, client_id, api_key_id, api_secret, public_key, private_key, c
             Private key used for crypto operations. Base64URL encoded string of
             bytes.
 
+        public_signing_key : str
+            Public signing key used for signing operations. 
+
+        private_signing_key : str
+            Private signing key used for signing operations. 
+
         client_email : str
             Email of the client.
             Optional.
 
         version : str
-            Version of the configuration file style. Defaults to 1.
+            Version of the configuration file style. Defaults to 2.
```

Comment:
Should we default to 2? or change to 2 when a signing key is present

Reply:
Now this is where we should have an if block, because they are lots of existing v1 clients out there and we want them to be able to continue to use the new sdk without breaking functionality or setting untruthful values (e.g. if there are no signing keys passed in to this init method by definition this is a v1 client)

Reply:
(to be clear, the parameter should still be a default of 2 for looking towards the future where we only have v2 clients or are only creating v2 clients, but that default is a facade of sorts because we still this if block to determine what version the client actually is based on presence of signing keys)

Reply:
Is it enough to just check the presence of the private signing key? Then it looks like 

```
if private_signing_key == None:
   self.version = 1
else:
   self.version = version # version parameter defaulting to 2 
```

Reply:
👍🏽 thats much more elegant!

Reply:
Are signing keys for v1 clients able to be set to empty strings before Config creation? They are not optional for Config init, so wanted to make that checking against an empty string is the correct thing to do